uc-crux-llvm/src/UCCrux/LLVM/Context/Function.hs

```diff
@@ -0,0 +1,275 @@
+{-
+Module       : UCCrux.LLVM.Context.Function
+Description  : LLVM function-level read-only context.
+Copyright    : (c) Galois, Inc 2021
+License      : BSD3
+Maintainer   : Langston Barrett <langston@galois.com>
+Stability    : provisional
+-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+module UCCrux.LLVM.Context.Function
+  ( FunctionContext (..),
+    SomeFunctionContext (..),
+    FunctionContextError (..),
+    ppFunctionContextError,
+    makeFunctionContext,
+    functionName,
+    moduleTypes,
+    argumentNames,
+    argumentCrucibleTypes,
+    argumentFullTypes,
+    argumentMemTypes,
+    argumentStorageTypes,
+  )
+where
+
+{- ORMOLU_DISABLE -}
```

Comment:
What are these comments about?

Reply:
I don't like taking the time to format code by hand, so I've used the [ormolu](https://hackage.haskell.org/package/ormolu) tool to format all the code in this package. However, `ormolu` will inexplicably choose to remove grouping of inputs, which I really don't like.

Reply:
Ah, that explains some of the formatting decisions.  I'm not always a huge fan of the way things are laid out, but I guess not enough to hold up this PR.

Does ormolu only run at file-level, or can you do it from your editor by selecting regions? I'm not generally a fan of leaving artifacts like this in files unless there's quite wide adoption of the tool among the developers.